contactaccessrights/tests/phpunit/CRM/Contactaccessrights/Helper/ContactActionsMenu/ContactAccessGroupTest.php

```diff
@@ -0,0 +1,188 @@
+use CRM_Contactaccessrights_Component_ContactActionsMenu_UserLocationsListItem as UserLocationsListItem;
+use CRM_Contactaccessrights_Component_ContactActionsMenu_UserACLGroupsListItem as UserACLGroupsListItem;
+
+/**
+ * Class CRM_Contactaccessrights_Helper_ContactAccessGroupTest
+ *
+ * @group headless
+ */
+class CRM_Contactaccessrights_Helper_ContactAccessGroupTest extends BaseHeadlessTest {
+
+  public function testMenuItemsAreCorrectlyAddedWhenContactIsAnAdmin() {
+    $contactUserInfo = ['cmsId' => 4, 'contact_id' => 5];
+    $aclGroups = [];
+    $userPermission = $this->prophesize(CMSUserPermission::class);
+    $userPermission->check($contactUserInfo, ['view all contacts', 'edit all contacts'])->willReturn(TRUE);
+    $contactRightsService = $this->prophesize(ContactRightsService::class);
+
+    $contactAccessActionGroup = new ContactAccessActionGroup(
+      $contactUserInfo,
+      $contactRightsService->reveal(),
+      $userPermission->reveal(),
+      $aclGroups
+    );
+
+    $contactAccessActionGroup = $contactAccessActionGroup->get();
+    $menuItems = $contactAccessActionGroup->getItems();
+    //Two Items are expected: The All staff Item and the Manage Regional Access Button
+    $this->assertCount(2, $menuItems);
+    $this->assertInstanceOf(GenericTextItem::class, $menuItems[0]);
+    $this->assertInstanceOf(ActionsGroupButtonItem::class, $menuItems[1]);
+
+    $this->assertEquals($contactAccessActionGroup->getTitle(), $this->getGroupTitle());
+  }
+
+  public function testMenuItemsAreCorrectlyAddedWhenContactIsNotAnAdminAndHasAccessRights() {
+    $contactUserInfo = ['cmsId' => 4, 'contact_id' => 5];
+    $aclGroups = ['Group 1'];
+    $userPermission = $this->prophesize(CMSUserPermission::class);
+    $userPermission->check($contactUserInfo, ['view all contacts', 'edit all contacts'])->willReturn(FALSE);
+    $contactRightsService = $this->prophesize(ContactRightsService::class);
+    $contactRightsService->getContactRightsByRegions($contactUserInfo['contact_id'])->willReturn([['label' => 'Region1']]);
+    $contactRightsService->getContactRightsByLocations($contactUserInfo['contact_id'])->willReturn([['label' => 'Location1']]);
+
+    $contactAccessActionGroup = new ContactAccessActionGroup(
+      $contactUserInfo,
+      $contactRightsService->reveal(),
+      $userPermission->reveal(),
+      $aclGroups
+    );
+
+    $contactAccessActionGroup = $contactAccessActionGroup->get();
+    $menuItems = $contactAccessActionGroup->getItems();
+    //Four Items are expected: Region List, Locations List, ACL Group
+    //List item and the Manage Regional Access Button
+    $this->assertCount(4, $menuItems);
+    $this->assertInstanceOf(UserRegionsListItem::class, $menuItems[0]);
+    $this->assertInstanceOf(UserLocationsListItem::class, $menuItems[1]);
+    $this->assertInstanceOf(UserACLGroupsListItem::class, $menuItems[2]);
+    $this->assertInstanceOf(ActionsGroupButtonItem::class, $menuItems[3]);
```

Comment:
mmmm... so this is probably why you've created the empty classes. I still don't think it's a good idea. I know we don't want to test the whole markup, so maybe we can call the render() method and check that the string contains the group, regions or location labels?

Reply:
Not only that, also incase we need to change the markup for a list item, we can do that with minimal change in the future. 
The render method would spit out html and I wonder how we would check that for the labels, hoping it won't be messy.

Reply:
You don't care about the markup. You only care if the string contains the label. Something like `assertContains('Region 1', $item->render())`